salt/modules/solarisips.py

```diff
@@ -301,7 +301,7 @@ def version(*names, **kwargs):
     if names:
         cmd = ['/bin/pkg', 'list', '-Hv']
         cmd.extend(names)
-        lines = __salt__['cmd.run_stdout'](cmd).splitlines()
+        lines = __salt__['cmd.run_stdout'](cmd, ignore_retcode=True).splitlines()
```

Comment:
Can you exmplain why you added this? `ignore_retcode` just suppresses Salt's normal behavior of logging an error (as well as the command's output) when a command returns a non-zero exit status. Does this command return a non-zero exit status in cases where there is no error?

Reply:
Yes, as described in the New Behavior of the PR : 
The version() function ignores the return code of pkg list to avoid errors when checking the version of non installed packages.

This is because lastest_version() calls version(). The package might not be installed and version will error out.